packages/@react-spectrum/s2/src/Tabs.tsx

```diff
@@ -10,9 +10,289 @@
+        return false;
+      }
+
+      // @ts-ignore
+      selectedKey = collection.getKeyAfter(selectedKey);
+      index++;
+    }
+  }
+
+  return true;
+}
+
+interface TabLineProps {
+  disabledKeys: Iterable<Key> | undefined,
+  isDisabled: boolean | undefined,
+  selectedTab: HTMLElement | undefined,
+  orientation?: Orientation
+}
+
+const selectedIndicator = style({
+  position: 'absolute',
+  color: {
+    default: 'neutral',
+    isDisabled: 'disabled'
+  },
+  height: size(2),
+  bottom: {
+    orientation: {
+      horizontal: 0
+    }
+  },
+  borderStyle: 'solid',
+  borderRadius: 'lg',
+  borderWidth: '[1px]',
+  boxSizing: 'border-box'
+});
+
+function TabLine(props: TabLineProps) {
+  let {
+    disabledKeys,
+    isDisabled: isTabsDisabled,
+    selectedTab,
+    orientation
+  } = props;
+  let {direction} = useLocale();
+  let state = useContext(TabListStateContext);
+
+  // We want to add disabled styling to the selection indicator only if all the Tabs are disabled
+  let [isDisabled, setIsDisabled] = useState<boolean>(false);
+  useEffect(() => {
+    let isDisabled = isTabsDisabled || isAllTabsDisabled(state?.collection, disabledKeys ? new Set(disabledKeys) : new Set(null));
+    setIsDisabled(isDisabled);
+  }, [state?.collection, disabledKeys, isTabsDisabled, setIsDisabled]);
 
-export function Tabs(props: TabsProps) {
-  return <RACTabs {...props} />;
+  let [style, setStyle] = useState<{transform: string | undefined, width: string | undefined, height: string | undefined}>({
+    transform: undefined,
+    width: undefined,
+    height: undefined
+  });
+
+  let onResize = useCallback(() => {
+    if (selectedTab) {
+      let styleObj: { transform: string | undefined, width: string | undefined, height: string | undefined } = { 
+        transform: undefined, 
+        width: undefined, 
+        height: undefined 
+      };
+
+      // In RTL, calculate the transform from the right edge of the tablist so that resizing the window doesn't break the Tabline position due to offsetLeft changes
+      let offset = direction === 'rtl' ? -1 * ((selectedTab.offsetParent as HTMLElement)?.offsetWidth - selectedTab.offsetWidth - selectedTab.offsetLeft) : selectedTab.offsetLeft;
+      styleObj.transform = orientation === 'vertical'
+        ? `translateY(${selectedTab.offsetTop}px)`
+        : `translateX(${offset}px)`;
+
+      if (orientation === 'horizontal') {
+        styleObj.width = `${selectedTab.offsetWidth}px`;
+      } else {
+        styleObj.height = `${selectedTab.offsetHeight}px`;
+      }
+      setStyle(styleObj);
+    }
+  }, [direction, setStyle, selectedTab, orientation]);
+
+  useLayoutEffect(() => {
+    onResize();
+  }, [onResize, state?.selectedKey, state, orientation]);
+  
+  return (
+    <div style={{...style}} className={selectedIndicator({isDisabled, orientation}) + ' ' + raw('transition: transform 130ms ease-in-out 0s, width 130ms ease-in-out 0s')} />
```

Comment:
There's no tokens for the animations when the selection indicator moves between the different tabs so I mostly guessed for these. Not sure if there's a way to add transitions without using raw?

Reply:
shouldn't need raw
see https://github.com/adobe/react-spectrum/blob/1dcc8705115364a2c2ead2ececae8883dd6e9d07/packages/%40react-spectrum/s2/src/ProgressBar.tsx#L105